modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/SemanticRequestMetrics.java

```diff
@@ -0,0 +1,118 @@
+
+import static com.spotify.apollo.StatusType.Family.INFORMATIONAL;
+import static com.spotify.apollo.StatusType.Family.SUCCESSFUL;
+import static java.util.Objects.requireNonNull;
+
+class SemanticRequestMetrics implements RequestMetrics {
+
+  private static final String COMPONENT = "service-request";
+  private final SemanticMetricRegistry metricRegistry;
+
+  private final MetricId fanoutId;
+  private final MetricId countRequestId;
+  private final MetricId timeRequestId;
+  private final Meter sentReplies;
+  private final Meter sentErrors;
+
+  SemanticRequestMetrics(
+      SemanticMetricRegistry metricRegistry,
+      MetricId id,
+      Meter sentReplies,
+      Meter sentErrors) {
+    this.metricRegistry = metricRegistry;
+    // Already tagged with 'service' and 'endpoint'. 'application' gets added by the ffwd reporter
+    Preconditions.checkArgument(id.getTags().containsKey("service"),
+                                "metricId must be tagged with 'service'");
+    Preconditions.checkArgument(id.getTags().containsKey("endpoint"),
+                                "metricId must be tagged with 'endpoint'");
+
+    MetricId metricId = id.tagged("component", COMPONENT);
+
+    fanoutId = metricId.tagged(
+        "what", "request-fanout-factor",
+        "unit", "request/request");
+
+    countRequestId = metricId.tagged(
+        "what", "endpoint-request-rate",
+        "unit", "request");
+
+    timeRequestId = metricId.tagged(
+        "what", "endpoint-request-duration");
+
+    this.sentReplies = requireNonNull(sentReplies);
+    this.sentErrors = requireNonNull(sentErrors);
+
+    registerRatioGauge(metricId, "1m", () -> RatioGauge.Ratio.of(this.sentErrors.getOneMinuteRate(),
+                                                                 this.sentReplies.getOneMinuteRate()));
+    registerRatioGauge(metricId, "5m", () -> RatioGauge.Ratio.of(this.sentErrors.getFiveMinuteRate(),
+                                                                 this.sentReplies.getFiveMinuteRate()));
+    registerRatioGauge(metricId, "15m", () -> RatioGauge.Ratio.of(this.sentErrors.getFifteenMinuteRate(),
+                                                                  this.sentReplies.getFifteenMinuteRate()));
+  }
+
+  private void registerRatioGauge(MetricId metricId,
+                                  String stat,
+                                  Supplier<RatioGauge.Ratio> ratioSupplier) {
+    metricRegistry.register(
+        metricId.tagged("what", "error-ratio", "stat", stat),
+        new RatioGauge() {
+          @Override
+          protected Ratio getRatio() {
+            return ratioSupplier.get();
+          }
+        });
+  }
+
+  @Override
+  public void fanout(int requests) {
+    metricRegistry.histogram(fanoutId).update(requests);
+  }
+
+  @Override
+  public void responseStatus(StatusType status) {
+    metricRegistry.meter(countRequestId.tagged("status-code", String.valueOf(status.code()))).mark();
```

Comment:
BTW; avoiding status-family, which we used internally because there are too many status codes, in favour of a configurable set of status codes to precreate. In general, for a given service, you will know which codes you're generating and it's often the case that you want the resolution individual codes give you.